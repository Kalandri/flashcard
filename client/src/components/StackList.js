import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { setStack, fetchStacks } from '../actions';

export class StackList extends Component {
  componentDidMount() {
    if (this.props.stacks.length === 0) this.props.fetchStacks();
  }

  render() {
    return (
      <div>
        <h3>List of Flash Cards</h3>
        {
          this.props.stacks.map(stack => {
            console.log(stack);
            return (
              <Link 
                to='/stack' 
                key={stack._id} 
                onClick={() => this.props.setStack(stack)}
              >
                <h4>{stack.title}</h4>
              </Link>
            )
          })
        }
      </div>
    )
  }
}

function mapStateToProps({stacks}) {
  return { stacks };
}

export default connect(mapStateToProps, { setStack, fetchStacks })(StackList);