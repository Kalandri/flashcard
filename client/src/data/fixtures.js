export const stack = {
  id: 0,
  title: 'test title 0',
  cards: [
    {id: 0, prompt:'test01', answer: 'answer01'},
	{id: 1, prompt:'test02', answer: 'answer02'}
  ]
};

export const stack1 = {
  id: 1,
  title: 'test title 1',
  cards: [
    {id: 0, prompt:'test11', answer: 'answer11'},
	{id: 1, prompt:'test12', answer: 'answer12'}
  ]
};

export const stacks = [stack, stack1];
