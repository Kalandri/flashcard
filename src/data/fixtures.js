export const stack = {
  id: 0,
  title: 'test title',
  cards: [
    {id: 0, prompt:'test1', answer: 'answer1'},
	{id: 1, prompt:'test2', answer: 'answer2'}
  ]
};
