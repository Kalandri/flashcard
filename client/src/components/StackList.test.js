import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { stacks } from '../data/fixtures';
import { StackList } from './StackList';

const props = { stacks };

describe('StackList', ()=>{
	const stackList = shallow(<StackList {...props} />);

	it('renders the correct number of the links', ()=>{
		expect(stackList.find('Link').length).toEqual(props.stacks.length);
	});
});

test('snapshot test for StackList', ()=>{
	const component = renderer.create(
		<MemoryRouter >
			<StackList {...props} />
		</MemoryRouter>
	);
	const tree=component.toJSON();
	expect(tree).toMatchSnapshot();
})