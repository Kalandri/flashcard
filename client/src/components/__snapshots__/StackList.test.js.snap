// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for StackList 1`] = `
<div>
  <a
    href="/stack"
    onClick={[Function]}
  >
    <h4>
      test title 0
    </h4>
  </a>
  <a
    href="/stack"
    onClick={[Function]}
  >
    <h4>
      test title 1
    </h4>
  </a>
</div>
`;
